### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by mrcbq**
In this opportunity, I did the following:

- [x] Make use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as backend.
- [x] Use [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make the HTTP requests to the backend
- [x] Fetch the books from the Bookstore API after the component has rendered
- [x] Refactor the add book and remove book features to persist the changes in the server
- [x] My application should behave the same way after the refactoring

### CODE REVIEW

File: src/components/Books.js [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
 
 import Book from './Book';
 import AddForm from './AddForm';
-import { addBook, deleteBook } from '../redux/books/booksSlice';
+import { fetchBooks, addBook, deleteBook } from '../redux/books/booksSlice';
 
 function BookList() {
-  const books = useSelector((state) => state.books);
+  const books = useSelector((state) => state.books.books);
+  // console.log('Books:', books);
```
Comment (t-yanick) [inline src/components/Books.js]: - [x] Great job working on this milestone. Kindly remove commented lines of codes from the codebase. This makes the codebase look clean and professional.

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @mrcbq , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  `createAsyncThunk` is imported in the project
:heavy_check_mark: HTTP requests are made with `axios`HTTP requests are made with [axios](https://axios-http.com/docs/intro)
:heavy_check_mark: `addBook` and `removeBook` are refactored accordingly
:heavy_check_mark: `addBook` and `removeBook` features  work the same as before.

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Books.js (hunk 1 above)
Comment (mrcbq) [reply src/components/Books.js]: It's done

**Review by ShadyShawkat — APPROVED** [review]
## Hi @mrcbq, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**